### PR TITLE
Bring in user followers from github api

### PR DESCRIPTION
**Pull request by tschaffer1618**
---FIRST PUSH---
-Build github_user class
-Add new method to github_service to pull in user followers from the API
-Add new method to user_facade to format the followers
-Add followers to user dashboard show method
-Update the user show view to display the followers

---SECOND PUSH---
-Clean up typos and formatting in tests
-Beef up model validations and relationships with tests
-Add github service tests using vcr cassettes
-Add github feature tests using webmock fixtures

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -4,9 +4,12 @@ class User < ApplicationRecord
   has_many :user_videos
   has_many :videos, through: :user_videos
 
-  validates :email, uniqueness: true, presence: true
-  validates_presence_of :password, allow_nil: true
+  validates_presence_of :email, uniqueness: true
+  validates_presence_of :password_digest
```
Comment (hsmitha26) [inline app/models/user.rb:8]: Line 8 makes a lot more sense than allow_nil: true.

File: Gemfile [hunk 2]
```diff
@@ -42,9 +42,12 @@ group :development, :test do
...
   gem 'shoulda-matchers'
   gem 'simplecov'
-  gem 'vcr'
   gem 'webdrivers'
+end
+
+group :test do
   gem 'webmock'
+  gem 'vcr'
 end
```
Comment (hsmitha26) [inline Gemfile]: Glad you moved these gems into the test environment, makes a lot more sense.

File: Gemfile [hunk 3]
```diff
@@ -42,9 +42,12 @@ group :development, :test do
   gem 'selenium-webdriver'
   gem 'shoulda-matchers'
   gem 'simplecov'
-  gem 'vcr'
   gem 'webdrivers'
+end
+
+group :test do
   gem 'webmock'
+  gem 'vcr'
```
Comment (hsmitha26) [inline Gemfile:50]: Glad you moved these gems into the test environment, makes a lot more sense.

File: spec/features/users/user_can_log_in_and_out_spec.rb [hunk 4]
```diff
@@ -41,7 +41,7 @@
 
     expect(current_path).to eq(root_path)
     expect(page).to_not have_content(user.first_name)
-    expect(page).to have_content('SIGN IN')
+    expect(page).to have_content('Sign In')
```
Comment (hsmitha26) [inline spec/features/users/user_can_log_in_and_out_spec.rb:44]: Glad we don't have this test failing.  Good catch!

File: spec/services/github_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
```
Comment (hsmitha26) [inline spec/services/github_service_spec.rb:1]: Thanks for making this test dynamic, I can add the feature where we can get GitHub users who are following  the logged in user.

**Comment by hsmitha26** [comment]
What are your thoughts on using the test, that you created, in place of the skipped one?
The test you created: spec/features/users/user_can_get_data_from_github_spec.rb

**Comment by hsmitha26** [comment]
Looking forward to discussing:
- Heroku deployment and functionality before adding the feature, User Dashboard: GitHub Following.

**Comment by tschaffer1618** [comment]
> What are your thoughts on using the test, that you created, in place of the skipped one?
> The test you created: spec/features/users/user_can_get_data_from_github_spec.rb

I think we will need both tests eventually- the new one assumes the user already has already connected to github and is testing if the user can consume the api. The skipped test is stubbing out the actually process of connecting to github.

**Pull request merged**